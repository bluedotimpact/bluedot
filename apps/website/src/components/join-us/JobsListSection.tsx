import { Card, Section } from '@bluedot/ui';
import { isMobile } from 'react-device-detect';
import { P } from '../Text';
import { CmsJobPosting } from '../../lib/api/db/tables';
import { ROUTES } from '../../lib/routes';

export type GetAshbyJobsResponse = {
  jobs: {
    id: string;
    title: string;
    department: string;
    team: string;
    employmentType: string;
    location: string;
    shouldDisplayCompensationOnJobPostings: boolean;
    publishedAt: string;
    isListed: boolean;
    isRemote: boolean;
    descriptionHtml: string;
  }[]
};

export type JobsListSectionProps = {
  ashbyJobs: GetAshbyJobsResponse['jobs']
  cmsJobs: Omit<CmsJobPosting, 'body'>[],
};

const JobsListSection = ({ ashbyJobs, cmsJobs }: JobsListSectionProps) => {
  const jobs = [
    ...cmsJobs.map((j) => ({ id: j.slug, title: j.title, location: j.subtitle })),
    ...ashbyJobs,
  ];

  return (
    <Section className="jobs-list-section" title="Careers at BlueDot Impact">
      <div id="open-roles-anchor" className="invisible relative bottom-48" />
      {jobs.length === 0 ? (
        <P>
          We're not currently running any open hiring rounds at the moment.
        </P>
      ) : (
        <div className="jobs-list__container flex flex-col gap-8">
          {jobs.map((job) => (
            <JobListItem key={job.id} job={job} />
          ))}
        </div>
      )}
    </Section>
  );
};

const JobListItem = ({ job }: {
  job: { id: string; title: string; location: string };
}) => {
  const url = `${ROUTES.joinUs.url}/${job.id}`;

  return (
    <div className="jobs-list__listing">
      <Card
        className="jobs-list__card container-lined hover:container-elevated p-8"
        ctaText="Learn more"
        ctaUrl={url}
        isEntireCardClickable={!isMobile}
        isFullWidth={!isMobile}
        subtitle={job.location}
        title={job.title}
      />
    </div>
  );
};

export default JobsListSection;
