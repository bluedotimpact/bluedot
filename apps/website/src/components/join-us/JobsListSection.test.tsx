import { render } from '@testing-library/react';
import {
  describe, expect, test, vi,
} from 'vitest';
import * as deviceDetect from 'react-device-detect';
import JobsListSection from './JobsListSection';

const mockCmsJobs = [
  {
    id: '1',
    slug: 'software-engineer',
    title: 'Software Engineer',
    subtitle: 'Remote',
    applicationUrl: 'https://example.com/apply/software-engineer',
    publicationStatus: 'Published' as const,
    publishedAt: Date.now() / 1000,
  },
  {
    id: '2',
    slug: 'product-manager',
    title: 'Product Manager',
    subtitle: 'London, UK',
    applicationUrl: 'https://example.com/apply/product-manager',
    publicationStatus: 'Published' as const,
    publishedAt: Date.now() / 1000,
  },
];

describe('JobsListSection', () => {
  test('renders default as expected', () => {
    const { container } = render(<JobsListSection jobs={mockCmsJobs} />);

    // Check if job titles and locations are rendered
    const content = container.textContent;
    expect(content).toContain('Software Engineer');
    expect(content).toContain('Product Manager');
    expect(content).toContain('Remote');
    expect(content).toContain('London, UK');
    expect(content).toContain('Learn more');

    // Check if there are two job listings
    const jobListings = container.querySelectorAll('.jobs-list__listing');
    expect(jobListings.length).toBe(2);
  });

  test('renders mobile as expected', () => {
    vi.spyOn(deviceDetect, 'isMobile', 'get').mockReturnValue(true);
    const { container } = render(<JobsListSection jobs={mockCmsJobs} />);

    // Check if job titles and locations are rendered
    const content = container.textContent;
    expect(content).toContain('Software Engineer');
    expect(content).toContain('Product Manager');
    expect(content).toContain('Remote');
    expect(content).toContain('London, UK');
    expect(content).toContain('Learn more');

    // Check if there are two job listings
    const jobListings = container.querySelectorAll('.jobs-list__listing');
    expect(jobListings.length).toBe(2);

    vi.restoreAllMocks();
  });

  test('renders empty state when no jobs are provided', () => {
    const { container } = render(<JobsListSection jobs={[]} />);
    expect(container.textContent).toContain("We're not currently running any open hiring rounds at the moment.");
  });
});
