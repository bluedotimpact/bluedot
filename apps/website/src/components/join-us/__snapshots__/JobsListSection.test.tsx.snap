// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`JobsListSection > renders empty state when no jobs are provided 1`] = `
<div>
  <section
    class="section section-body jobs-list-section"
  >
    <div
      class="section-heading__title-container flex justify-between items-center gap-space-between mb-6"
    >
      <div
        class="section-heading__content flex-1 flex flex-col gap-2"
      >
        <h2
          class="section-heading__title relative ml-[-0.08em] bluedot-h2"
        >
          Careers at BlueDot Impact
        </h2>
      </div>
    </div>
    <div
      class="section__body"
    >
      <div
        class="invisible relative bottom-48"
        id="open-roles-anchor"
      />
      <p
        class="bluedot-p not-prose"
      >
        We're not currently running any open hiring rounds at the moment.
      </p>
    </div>
  </section>
</div>
`;

exports[`JobsListSection > renders mobile as expected 1`] = `
<div>
  <section
    class="section section-body jobs-list-section"
  >
    <div
      class="section-heading__title-container flex justify-between items-center gap-space-between mb-6"
    >
      <div
        class="section-heading__content flex-1 flex flex-col gap-2"
      >
        <h2
          class="section-heading__title relative ml-[-0.08em] bluedot-h2"
        >
          Careers at BlueDot Impact
        </h2>
      </div>
    </div>
    <div
      class="section__body"
    >
      <div
        class="invisible relative bottom-48"
        id="open-roles-anchor"
      />
      <div
        class="jobs-list__container flex flex-col gap-8"
      >
        <div
          class="jobs-list__listing"
        >
          <div
            class="card flex items-start transition-transform duration-200 flex-col jobs-list__card container-lined"
          >
            <div
              class="card__content flex gap-6 w-full flex-1 justify-between flex-col"
            >
              <div
                class="card__text"
              >
                <p
                  class="card__title bluedot-h4 mb-2"
                >
                  Software Engineer
                </p>
                <p
                  class="card__subtitle bluedot-p "
                >
                  Full-time position
                </p>
              </div>
              <div
                class="card__bottom-section flex flex-col gap-space-between"
              >
                <a
                  class="cta-button flex items-center justify-center rounded-sm transition-all duration-200 text-sm px-4 py-3 w-fit font-[650] whitespace-nowrap cursor-pointer not-prose cta-button--secondary bg-transparent border border-bluedot-normal text-bluedot-normal hover:bg-bluedot-lighter card__cta"
                  href="/join-us/software-engineer"
                  tabindex="0"
                >
                  <span
                    class="cta-button__text"
                  >
                    Learn more
                  </span>
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="jobs-list__listing"
        >
          <div
            class="card flex items-start transition-transform duration-200 flex-col jobs-list__card container-lined"
          >
            <div
              class="card__content flex gap-6 w-full flex-1 justify-between flex-col"
            >
              <div
                class="card__text"
              >
                <p
                  class="card__title bluedot-h4 mb-2"
                >
                  Product Manager
                </p>
                <p
                  class="card__subtitle bluedot-p "
                >
                  Full-time position
                </p>
              </div>
              <div
                class="card__bottom-section flex flex-col gap-space-between"
              >
                <a
                  class="cta-button flex items-center justify-center rounded-sm transition-all duration-200 text-sm px-4 py-3 w-fit font-[650] whitespace-nowrap cursor-pointer not-prose cta-button--secondary bg-transparent border border-bluedot-normal text-bluedot-normal hover:bg-bluedot-lighter card__cta"
                  href="/join-us/product-manager"
                  tabindex="0"
                >
                  <span
                    class="cta-button__text"
                  >
                    Learn more
                  </span>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
